site-src/geps/gep-1364.md

```diff
@@ -0,0 +1,389 @@
+a map that looks like a list in YAML form).
+3. Each has a number of fields, the most important of which for this discussion
+are `type`, `status`, `reason`, and `observedGeneration`.
+
+    * `type` is a string value indicating the Condition type. `Accepted`, `Scheduled`,
+    and `Ready` are current examples.
+    * `status` indicates the state of the condition, and can be one of three values,
+    `true`, `false`, or `unknown`. Unkown in particular is important, because it
+    means that the controller is unable to determine the status for some reason.
+    * `reason` is a CamelCase string that is a brief description of the reason why
+    the `status` is set the way it is.
+    * `observedGeneration` is an optional field that sets what the `metadata.generation`
+    field was when the controller last saw a resource.
+
+4. Conditions shoud describe the _current state_ of the resource at observation
+time, which means that they should be an adjective (like `Ready`), or a past-tense
+verb (like `Attached`).
+5. Conditions should be applied to a resource the first time the controller sees
+the resource. This seems to imply that _all conditions should be present on every
+resource owned by a controller_, but the rest of the conventions don't make this
+clear, and it is often not complied with.
+6. It's helpful to have a top-level condition which summarizes more detailed conditions.
+The guidelines suggest using either `Ready` for long-running processes, or `Succeeded`
+for bounded exection.
+
+From these guidelines, we can see that Conditions can be either _positive polarity_
+(healthy resources have them as `status: true`) or _negative polarity_ (healthy
+resources have them as `status: false`). `Ready` is an example of a positive polarity
+condition, and conditions like `Conflicted` from Listener or `NetworkUnavailable`,
+`MemoryPressure`, or `DiskPressure` from the Node resource are examples of
+negative-polarity conditions.
+
+There is also some extra context that's not in the API conventions doc:
+
+SIG-API Machinery has been reluctant to add fields that would aid in machine-parsing
+of Conditions, especially fields that would indicate the polarity, because they
+are intended more for human consumption than machine consumption.
+
+This means that there's no guidance from upstream about condition polarity. We'll
+discuss this more when we talk about new conditions.
+
+The guidance about Conditions being added as soon as a controller sees a resource
+is a bit unclear - as written in the conventions, it seems to imply that _all_ 
+relevant conditions should always be added, even if their status has to be set to
+`unknown`.
+Gateway API resources do not currently require this, and the practice seems to be
+uncommon.
+
+## Proposed changes
+
+### Proposed changes summary
+
+* All the current Conditions that indicate that the resource is okay and ready
+for processing witll be replaced with `Attached`, except for GatewayClass (since
```

Comment:
Sound good - but I think Gateway may need additional info, like Ready - since the process may involve 2 steps.
For example Attached will be set when the controller generates the configs - but Ready when certificates are provisioned. CertManager would watch for Attached and start the ACME process, create the Secret - and the gateway controller would switch to Ready.

Reply:
I'm proposing a `Programmed` condition to summarize this without the "this is ready right now" implications of Ready. I'll put those changes into this summary.